app/models/group.rb

```diff
@@ -369,8 +335,7 @@ def discussion_private_default
 
   def add_member!(user, inviter=nil)
     begin
-      save!
-      memberships.find_or_create_by(user: user) { |m| m.inviter = inviter }
+      tap(&:save).memberships.find_or_create_by(user: user) { |m| m.inviter = inviter }
```

Comment:
Good call. Not to mention we definitely don't want to be trying to add members to an invalid group, which seems like a really scary path to be going down.